kotlin/android/src/main/java/com/reactnativesecurekeystore/SecureKeystoreImpl.kt

```diff
@@ -34,18 +38,30 @@ class SecureKeystoreImpl(
 
     /** Generate a new key pair */
     override fun generateKeyPair(
-      alias: String,
-      isAuthRequired: Boolean,
-      authTimeout: Int?,
+        type: String,
+        alias: String,
+        isAuthRequired: Boolean,
+        authTimeout: Int?,
     ): String {
-        val keyPair = keyGenerator.generateKeyPair(alias, isAuthRequired, authTimeout)
+        val keyPair: KeyPair
+        keyPair = if (type == "RS256") {
+            keyGenerator.generateKeyPair(alias, isAuthRequired, authTimeout)
+        } else
```

Comment:
can if handle it like, 
if type is EC, generate EC key pair else throw exception with not supported type